src/cars/car.service.ts

```diff
@@ -1,82 +1,105 @@
-import { Car } from './entities/car.entity';
+import { CarDto } from './dto';
+import { ERROR_UNIQUE_VALUE } from 'consts';
+import { Car } from '@prisma/client';
 
 @Injectable()
 export class CarsService {
-  constructor(@InjectRepository(Car) private carsRepository: Repository<Car>) {}
+  constructor(private prisma: PrismaService) {}
 
-  async getAll(user: User, searchCarDto: SearchCarDto): Promise<Car[]> {
-    const query = this.carsRepository.createQueryBuilder('car').where({ user });
-    const { search } = searchCarDto;
-
-    if (search) {
-      query.andWhere(
-        '(LOWER(car.name) LIKE LOWER(:search) OR LOWER(car.numberPlate) LIKE LOWER(:search) OR LOWER(car.date) LIKE LOWER(:search) OR LOWER(car.distance) LIKE LOWER(:search))',
-        { search: `%${search}%` },
+  numberPlateNotFound = (car: Car, numberPlate: string) => {
+    if (!car) {
+      throw new NotFoundException(
+        `The number plate:${numberPlate} doesn't exist.`,
       );
     }
-    try {
-      const cars = await query.getMany();
-      return cars;
-    } catch (error) {
-      throw new InternalServerErrorException();
+    return true;
+  };
+
+  numberPlateDuplicated = (error: any, numberPlate: string) => {
+    if (error === ERROR_UNIQUE_VALUE) {
+      throw new ConflictException(
+        `The number plate: ${numberPlate} already exists.`,
+      );
     }
+    return true;
+  };
+
+  async getCars(userId: number) {
+    return await this.prisma.car.findMany({
+      where: {
+        userId,
+      },
+    });
   }
 
-  async getByNumberPlate(numberPlate: string, user: User): Promise<Car> {
-    const car = await this.carsRepository.findOne({
-      where: { numberPlate, user },
+  async getCarByNumberPlate(userId: number, numberPlate: string) {
+    const car = await this.prisma.car.findFirst({
+      where: {
+        numberPlate: numberPlate,
+        userId,
+      },
     });
-    if (!car) {
-      throw new NotFoundException(
-        `Car with number plate ${numberPlate} not found.`,
-      );
+
+    if (this.numberPlateNotFound(car, numberPlate)) {
+      return car;
     }
-    return car;
   }
 
-  async create(createCarDto: CreateCarDto, user: User): Promise<void> {
-    const car = this.carsRepository.create({ ...createCarDto, user });
-    const { numberPlate } = createCarDto;
+  async createCar(userId: number, carDto: CarDto) {
+    const { numberPlate } = carDto;
+
     try {
-      await this.carsRepository.save(car);
+      return await this.prisma.car.create({
+        data: {
+          userId: userId,
+          ...carDto,
+        },
+      });
     } catch (error) {
-      if (error.code === ERROR_DUPLICATE_KEY_VALUE) {
-        throw new ConflictException(
-          `The number plate: ${numberPlate} already exists.`,
-        );
-      } else {
-        throw new InternalServerErrorException();
-      }
+      this.numberPlateDuplicated(error.code, numberPlate);
+      throw new InternalServerErrorException();
```

Comment:
If we are using such final error, we should consider storing error code somewhere (db/file/pass to logger).